### ISSUE TITLE
Execs are not scheduled by default

### ISSUE THREAD

**Issue by MattRiesterer** [christiangalsterer/execbeat#6]
The documentation states that 

```
# Optional cron expression, defines when to execute the command.
# Default is every 1 minute.
#cron:
```
https://github.com/christiangalsterer/execbeat/blob/master/execbeat.yml

Using this configuration did not schedule any execs. Only after setting the `cron` value specifically like 

```
cron: "@every 10s"
```

triggered the execs as expected.

Occurred with execbeat v2.1.1 and v2.2.0 using the amd64.deb package on Linux.

As most people will try first using the provided configuration, you might consider fixing this. It took me a while to figure this out.

Keep up the great work!

thanks a lot

Matthias

**Comment by christiangalsterer** [comment]
Hi Matthias,

thanks raising this. I'm currently travelling and will look into it in about a week from now.

**Comment by christiangalsterer** [comment]
I just tried it and it works for me. The default is one minute, i.e. you have to wait one minute after starting the beat until you see any output.
Did you wait for a minute and/or can check again?

**Comment by christiangalsterer** [comment]
Hi Matthias,

did you had a chance to check again of the scheduler is triggered after a minute has passed?

**Comment by MattRiesterer** [comment]
Hi Christian,

I'm sorry for replying this late, I somehow haven't seen the previous github notifications.

I can confirm that having `cron`commented out does not trigger the exec periodically.

**config**

```yaml
execbeat:

execs:
    -
      command: echo
      args: "hello"
      #cron: "@every 10s"
```

**log**
```
# tail -f /var/log/execbeat/execbeat
2017-02-18T08:16:03+01:00 INFO Flush Interval set to: 1s
2017-02-18T08:16:03+01:00 INFO Max Bulk Size set to: 2048
2017-02-18T08:16:03+01:00 DBG  create bulk processing worker (interval=1s, bulk size=2048)
2017-02-18T08:16:03+01:00 INFO execbeat start running.
2017-02-18T08:16:03+01:00 INFO execbeat is running! Hit CTRL-C to stop it.
2017-02-18T08:16:03+01:00 DBG  Creating poller # 0 with command: echo
2017-02-18T08:16:03+01:00 INFO Metrics logging every 30s
2017-02-18T08:16:33+01:00 INFO No non-zero metrics in the last 30s
2017-02-18T08:17:03+01:00 INFO No non-zero metrics in the last 30s
2017-02-18T08:17:33+01:00 INFO No non-zero metrics in the last 30s
2017-02-18T08:18:03+01:00 INFO No non-zero metrics in the last 30s
```

So the poller started at 08:16:03 but did not emit any message until 08:18:03. So I waited two minutes.

while with config
```yaml
execs:
    -
      command: echo
      args: "hello"
      cron: "@every 10s"
```

log shows

```
2017-02-18T08:19:05+01:00 DBG  Disable stderr logging
2017-02-18T08:19:05+01:00 INFO Home path: [/usr/share/execbeat] Config path: [/etc/execbeat] Data path: [/var/lib/execbeat] Logs path: [/var/log/execbeat]
2017-02-18T08:19:05+01:00 INFO Setup Beat: execbeat; Version: 2.2.0
2017-02-18T08:19:05+01:00 INFO Max Retries set to: 3
2017-02-18T08:19:05+01:00 INFO Activated logstash as output plugin.
2017-02-18T08:19:05+01:00 DBG  Create output worker
2017-02-18T08:19:05+01:00 DBG  No output is defined to store the topology. The server fields might not be filled.
2017-02-18T08:19:05+01:00 INFO Flush Interval set to: 1s
2017-02-18T08:19:05+01:00 INFO Max Bulk Size set to: 2048
2017-02-18T08:19:05+01:00 DBG  create bulk processing worker (interval=1s, bulk size=2048)
2017-02-18T08:19:05+01:00 INFO execbeat start running.
2017-02-18T08:19:05+01:00 INFO execbeat is running! Hit CTRL-C to stop it.
2017-02-18T08:19:05+01:00 DBG  Creating poller # 0 with command: echo
2017-02-18T08:19:05+01:00 INFO Metrics logging every 30s
2017-02-18T08:19:15+01:00 DBG  Executing command: [echo] with args [%!w(string=hello)]
2017-02-18T08:19:15+01:00 DBG  Error trying to unmarshal %!s(<nil>)
2017-02-18T08:19:15+01:00 DBG  Publish: {
  "@timestamp": "2017-02-18T07:19:15.001Z",
  "beat": {

...
```

The poller started at 08:19:05 and emitted the first message at 08:19:15. That's exactly 10 seconds after the start as configured.

I'm running this example on a debian64 vagrant box (for development purposes). Pretty standard. Not sure what else I can provide to help you to dig into the issue. If there's something, please just let me know.

Thank you and best regards,

matthias

**Comment by christiangalsterer** [comment]
Hi Matthias,

no problem ;-)
I was able to reproduce the problem now with the deb package on a Ubuntu Server 16.10.

**Comment by christiangalsterer** [comment]
I also tried the tar.gz version on Ubuntu and also there the default triggering is not working. On a mac it works without any issues. Need to further look into this.

**Comment by christiangalsterer** [comment]
I finally found the issue. It will be fixed with the next version 3.0.0 which I plan to release in a few days.

**Comment by christiangalsterer** [comment]
Just release version 3.0.0. Please note the release notes as there are some configuration changes.
